resources/functions/user-management/index.py

```diff
@@ -3,112 +3,91 @@
 
 import json
 import os
-import utils
+from http import HTTPStatus
 from aws_lambda_powertools import Tracer
 from aws_lambda_powertools import Logger
 from aws_lambda_powertools.logging import correlation_paths
 from aws_lambda_powertools.event_handler import APIGatewayRestResolver
-import idp_object_factory
+from cognito_user_management_service import CognitoUserManagementService
 
 tracer = Tracer()
 logger = Logger()
 app = APIGatewayRestResolver()
 
 idp_name = os.environ['IDP_NAME']
 idp_details=json.loads(os.environ['IDP_DETAILS'])
-
-idp_user_mgmt_service = idp_object_factory.get_idp_user_mgmt_object(idp_name)
+idp_user_mgmt_service = CognitoUserManagementService()
 
 @app.post("/users")
 @tracer.capture_method
 def create_user():
     user_details = app.current_event.json_body
-    logger.info("Request received to create new user")
     user_details['idpDetails'] = idp_details
     response = idp_user_mgmt_service.create_user(user_details)
-    logger.info("Request completed to create new user ")
-
-    return utils.create_success_response("New user created")
+    logger.info(response)    
+    return {"response": "New user created"}, HTTPStatus.OK.value
 
 @app.get("/users")
 @tracer.capture_method
 def get_users():
-    users = []
     user_details = {}
-    user_details['idpDetails'] = idp_details  
-    
-    logger.info("Request received to get user")
-    response = idp_user_mgmt_service.get_users(user_details)
-        
-    logger.info(response) 
-    num_of_users = len(response)
-    return utils.generate_response(response)
-
+    user_details['idpDetails'] = idp_details
+    users = idp_user_mgmt_service.get_users(user_details)
+    logger.info(users.serialize())    
+    return users.serialize(), HTTPStatus.OK.value
 
 @app.get("/users/<username>")
 @tracer.capture_method
 def get_user(username):
     user_details = {}
     user_details['idpDetails'] = idp_details
     user_details['userName'] = username
+    user = idp_user_mgmt_service.get_user(user_details)
+    if (user == None):
+        logger.info("User not found")        
+        return {"response": "User not found"}, HTTPStatus.NOT_FOUND.value
```

Comment:
powertools has these helper functions you can use: https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#raising-http-errors

https://github.com/awslabs/sbt-aws/blob/2463b9a751223aca90b6e4d95df374ca11342e97/resources/functions/tenant-management/tenant_management.py#L87